src/client/Microsoft.Identity.Client/ApiConfig/Executors/ConfidentialClientExecutor.cs

```diff
@@ -57,6 +57,8 @@ public ConfidentialClientExecutor(IServiceBundle serviceBundle, ConfidentialClie
                 _confidentialClientApplication.AppTokenCacheInternal);
 
             requestParams.SendX5C = clientParameters.SendX5C;
+            requestParams.AutoDetectRegion = clientParameters.AutoDetectRegion;
+            requestContext.ServiceBundle.Config.AuthorityInfo.AutoDetectRegion = clientParameters.AutoDetectRegion;
```

Comment:
We should throw a nice exception if user tries to use this flag with ADFS authrority.